src/sio.c

```diff
@@ -0,0 +1,259 @@
+/* SPDX-License-Identifier: MIT */
+
+#include "adt.h"
+#include "errno.h"
+#include "malloc.h"
+#include "sio.h"
+#include "types.h"
+#include "utils.h"
+
+// Reuse pages for different data sections, if space allows it
+#define MERGE_SIO_FWDATA
+
+#define SIO_KEY(s)  _SIO_KEY(#s)
+#define _SIO_KEY(s) (((s)[0] << 24) | ((s)[1] << 16) | ((s)[2] << 8) | (s)[3])
+
+#define MAX_FWDATA   8
```

Comment:
this is in conflict with the maximum of 8 `memory-region` in the proposed dt-binding since 2 `memory-region`s are used for firmware text and data

Reply:
Right